app/uk/gov/hmrc/exports/services/NotificationService.scala

```diff
@@ -178,4 +194,24 @@ class NotificationService @Inject()(submissionRepository: SubmissionRepository,
     if (exportsPointer.isEmpty) logger.warn(s"Missing pointer mapping for [${wcoPointer}]")
     exportsPointer
   }
+
+  def reattemptParsingUnparsedNotifications(): Future[Unit] =
+    notificationRepository
+      .findUnparsedNotifications()
+      .map(_.foreach { unparsedNotification =>
+        try {
+          val notifications = parseNotificationsPayload(unparsedNotification.actionId, XML.loadString(unparsedNotification.payload))
+
+          //successfully parsed the previously unparsable notification
+          notifications.head.details.map { _ =>
+            notificationRepository.removeNotificationsForActionId(unparsedNotification.actionId)
+            saveAll(notifications)
```

Comment:
What do you think about swaping the order for these operations? To save first and then to remove the old one? Or even better, to execute them both as a single bulk operation?
The reason for this is that in the current order we remove the old Notification before making sure new ones are persisted.

Reply:
Yes this is a good spot! I've altered my code to save first and delete after, this way it is only vulnerable to creating duplicate notifications on system crashes. 

I've been looking at the current code and it seems there is currently an opportunity for notifications to be duplicated in the event of a system crash and re-delivery. The batch idea is a good one to prevent this but would mean reworking the existing notification controller logic also. Maybe a tech debt ticket in the making?